pkg/controllermanager/controller/shoot/add.go

```diff
@@ -45,6 +46,7 @@ func AddToManager(mgr manager.Manager, cfg config.ControllerManagerConfiguration
 
 	if err := (&maintenance.Reconciler{
 		Config: cfg.Controllers.ShootMaintenance,
+		Clock:  clock.RealClock{},
```

Comment:
Please default this inside `AddToManager` function.
ref: https://github.com/gardener/gardener/pull/7062/commits/7b2ccf86aed92e42cce2109759a0abc5df8babe3